### PR TITLE
SimpleSAML_XML_* classes refactorized to PSR-4 and PSR-2

### PR DESCRIPTION
**Pull request by sgomez**
Two commits: one refactorized all classes in lib/SimpleSAML/XML directory to PSR-4 and the other to PSR-2.

Some new tests added.

### CODE REVIEW

**Comment by jaimeperez** [comment]
¡Gracias Sergio!

This is incredibly useful 😉 

I had a quick look right now at it, but it's a bit difficult to view the actual changes in the diffs here in github. From what I see, I gather it's only changes to the name classes and formatting, no functional changes, right?

**Comment by sgomez** [comment]
Hi Jaime!

Yes, I split up the changes in two differents commits, because PSR-2 changes are too many. PSR-4 is just adding namespaces and import external classes with `use`.

PSR-2 was done automatically with php-cs-fixer and PSR-2 rules. It's a very tested tools so there should be no problems.

File: lib/SimpleSAML/XML/Errors.php [hunk 1]
```diff
@@ -98,8 +101,8 @@ public static function end() {
 	 *
 	 * This function formats the given LibXMLError object as a string.
 	 *
-	 * @param $error  The LibXMLError which should be formatted.
-	 * @return  A string representing the given LibXMLError.
+	 * @param $error \LibXMLError  The LibXMLError which should be formatted.
```
Comment (jaimeperez) [inline lib/SimpleSAML/XML/Errors.php]: Shouldn't this be the other way around?

```
 * @param \LibXMLError $error ...
```

I see this happened all over the code, so we should fix it.

File: lib/SimpleSAML/XML/Shib13/AuthnResponse.php [hunk 2]
```diff
@@ -6,12 +6,23 @@
...
+use SimpleSAML\Utils\Config;
+use SimpleSAML\Utils\Random;
+use SimpleSAML\Utils\Time;
+use SimpleSAML\XML\Validator;
+
+class AuthnResponse {
 
 	/**
-	 * This variable contains an XML validator for this message.
+	 * @var Validator This variable contains an XML validator for this message.
```
Comment (jaimeperez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: I believe here we should use the full name of the class, including the namespace. I've observed problems with documentation that doesn't do that, because the `use` clauses in the beginning of the file are ignored. Therefore, I think we should use:

```
 * @var SimpleSAML\XML\Validator ...
```

File: lib/SimpleSAML/XML/Shib13/AuthnResponse.php [hunk 3]
```diff
@@ -260,9 +273,9 @@ public function getIssuer() {
 		$nodelist = $this->doXPathQuery($query);
 
 		if ($attr = $nodelist->item(0)) {
-			return $attr->value;
+			return $attr->nodeValue;
```
Comment (jaimeperez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: Was this a bug?

Comment (sgomez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: Yes, but it's my fault. I change it because I though it is a \DOMElement but it is a \DOMAttr. I tested it and it works fine because internally must be equivalent (both extends from DOMNode). I'm going to revert the change.

File: lib/SimpleSAML/XML/Shib13/AuthnResponse.php (hunk 2 above)
Comment (sgomez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: Ok, I'm going to check all this phpdoc issues in a new commit.

File: lib/SimpleSAML/XML/Shib13/AuthnResponse.php [hunk 4]
```diff
@@ -131,10 +125,11 @@ public function validate()
     }
 
 
-    /* Checks if the given node is validated by the signature on this response.
+    /**
+     * Checks if the given node is validated by the signature on this response.
      *
-     * Returns:
-     *  TRUE if the node is validated or FALSE if not.
+     * @param \DOMElement Node to be validated.
```
Comment (jaimeperez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: The name of the parameter here (`$node`) is missing. I see phpdoc supports that, but they discourage the practice. It's not really a big issue, specially given this is a private method, but it would be nice to correct it 😊

Comment (sgomez) [inline lib/SimpleSAML/XML/Shib13/AuthnResponse.php]: Sorry, I trust too much on the IDE syntax highlight. We missed this.

**Comment by jaimeperez** [comment]
No problem Sergio! I do that too... 😉 

Merged! Thanks a lot again!
